### DISCUSSION TITLE
Having trouble cloning GitHub source code into Discover system

### DISCUSSION THREAD

**Discussion by evanr1232** [NASA-LIS/LISF#1284] [Q&A] [answered]
Hello, 

I'm currently following the "LISF on NCCS Discover: Quick Start Guide", and working on cloning the source code. It looks like the guide wants me to clone everything into my local system, but when I go to the next step of the guide, "Compiling LISF Components", it looks like I should've done these cloning steps in the Discover environment, and I get errors when I try to compile. I'm just confused as to what the correct way to clone the source code is.

Specifically, the error I'm getting when I try to compile is:

Setting up configuration for LIS
--------------ERROR---------------------
Please specify the LIS architecture option 
and linker using the LIS_ARCH variable.
Configuration exiting ....
--------------ERROR---------------------
mv: cannot stat 'configure.lis': No such file or directory
mv: cannot stat 'LIS_misc.h': No such file or directory
mv: cannot stat 'LIS_NetCDF_inc.h': No such file or directory
Settings are written to configure.lis in the make directory.
If you wish to change settings, please edit that file.

To compile, run the compile script.

Thank you so much!

**Comment by dmocko** [comment, marked as answer]
Hi @evanr1232 

It seems that you have not properly set up your module file on Discover.  The module file will specify the LIS_ARCH variable, and other environmental variables needed for the configuration/compilation.  Please follow the steps in the Discover Quick Start Guide on how to ensure that you have loaded the module correctly to configure/compile the LISF software.

For more of a walkthrough on how to clone to code onto Discover, see this page (particularly Step 4):
https://github.com/NASA-LIS/LISF/blob/master/docs/working_with_github/working_with_github.adoc
